src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -5485,6 +5549,26 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             }
             tmp_vec.push_back(ASR::make_Assignment_t(al, x.base.base.loc, target, tmp_value,
                                     overloaded));
+            if ( target->type == ASR::exprType::Var &&
+                    tmp_value->type == ASR::exprType::StructConstructor ) {
+                Vec<ASR::call_arg_t> new_args; new_args.reserve(al, 1);
+                ASR::call_arg_t self_arg;
+                self_arg.loc = x.base.base.loc;
+                ASR::symbol_t* st = ASR::down_cast<ASR::Var_t>(target)->m_v;
+                self_arg.m_value = target;
+                new_args.push_back(al,self_arg);
+                AST::Call_t* call = AST::down_cast<AST::Call_t>(x.m_value);
+                if ( call->n_keywords>0 ) {
+                    throw SemanticError("Kwargs not implemented yet", x.base.base.loc);
+                }
+                visit_expr_list(call->m_args, call->n_args, new_args);
+                ASR::symbol_t* der = ASR::down_cast<ASR::StructType_t>(
+                    ASR::down_cast<ASR::Variable_t>(st)->m_type)->m_derived_type;
+                std::string call_name = "__init__";
+                ASR::symbol_t* call_sym = get_struct_member(der, call_name, x.base.base.loc);
+                tmp_vec.push_back(make_call_helper(al, call_sym,
```

Comment:
This modifies the meaning of the code, I mean, we have to handle this in the ASR pass.
The generated ASR must be the same as the original source code. 
This can be done in the upcoming PR.

Reply:
Example
```py
x: Class = Class(1, 3)
```

This PR creates the following
```py
x: Class
x__init__(x, 1, 3)
```
But, it has to be
```
x: Class
x = Class(1, 3)
```
and handle the above in the ASR pass